libnd4j/include/ops/declarable/platform/cudnn/conv2d.cu

```diff
@@ -289,6 +289,15 @@ PLATFORM_CHECK(conv2d, ENGINE_CUDA) {
     const bool badBiasType    = bias == nullptr ? false : (bias->dataType() != DataType::DOUBLE && bias->dataType() != DataType::FLOAT32 && bias->dataType() != DataType::HALF);
 
     return paddingMode != 2 && !badInputType && !badWeightsType && !badBiasType;
+    Requirements req("CUDNN CONV2d OP");
+    req.expectNotEq(makeInfoVariable(paddingMode,"paddingMode"), 2) &&
+    req.expectIn(makeInfoVariable(input->dataType(), TYPE_MSG_INPUT0), {DataType::HALF, DataType::FLOAT32, DataType::DOUBLE} ) &&
+    req.expectIn(makeInfoVariable(weights->dataType(), TYPE_MSG_INPUT1), {DataType::HALF, DataType::FLOAT32, DataType::DOUBLE} );
```

Comment:
For performance reasons, I would imagine us wanting all of this behind an if debug. Do you mind implementing this?